src/main/java/org/sagebionetworks/bridge/spring/controllers/BaseController.java

```diff
@@ -308,6 +309,17 @@ CriteriaContext getCriteriaContext(UserSession session) {
             .withStudyIdentifier(session.getStudyIdentifier())
             .build();
     }
+    
+    protected @Nonnull <T> T parseJson(TypeReference<? extends T> clazz) {
+        try {
+            return MAPPER.readValue(request().getInputStream(), clazz);
+        } catch (Throwable ex) {
+            if (Throwables.getRootCause(ex) instanceof InvalidEntityException) {
+                throw (InvalidEntityException)Throwables.getRootCause(ex);
+            }
+            throw new InvalidEntityException("Error parsing JSON in request body: " + ex.getMessage());    
+        }
+    }
```

Comment:
In BaseController.Test, we have tests for parseJson(Class). Should we add unit tests for parseJson(TypeReference)

Reply:
Added tests